aptos-move/framework/src/natives/string_utils.rs

```diff
@@ -0,0 +1,322 @@
+                && type_.address == AccountAddress::ONE
+            {
+                let mut v = strct
+                    .unpack()?
+                    .next()
+                    .unwrap()
+                    .value_as::<Vector>()?
+                    .unpack_unchecked()?;
+                if v.is_empty() {
+                    out.push_str("None");
+                } else {
+                    out.push_str("Some(");
+                    let inner_ty = if let MoveTypeLayout::Vector(inner_ty) = &fields[0].layout {
+                        inner_ty.deref()
+                    } else {
+                        unreachable!()
+                    };
+                    native_format_impl(context, base_gas, inner_ty, v.pop().unwrap(), out)?;
+                    out.push(')');
+                }
+                return Ok(());
+            }
+            write!(out, "{} {{", type_.name.as_str()).unwrap();
+            format_vector_with_fields(context, base_gas, fields, strct, out)?;
+            out.push('}');
+        },
+        MoveTypeLayout::Struct(MoveStructLayout::WithFields(fields)) => {
+            let strct = val.value_as::<Struct>()?;
+            out.push('{');
+            format_vector_with_fields(context, base_gas, fields, strct, out)?;
+            out.push('}');
+        },
+        MoveTypeLayout::Struct(MoveStructLayout::Runtime(fields)) => {
+            let strct = val.value_as::<Struct>()?;
+            out.push('{');
+            format_vector(
+                context,
+                base_gas,
+                fields.iter().zip(strct.unpack()?).collect(),
+                out,
+            )?;
+            out.push('}');
+        },
+    };
+    Ok(())
+}
+
+pub fn native_format(
+    gas_params: &GasParameters,
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert!(ty_args.len() == 1);
+    let ty = context
+        .deref()
+        .type_to_fully_annotated_layout(&ty_args[0])?
+        .unwrap();
+    let x = safely_pop_arg!(arguments, Reference);
+    let v = x.read_ref().map_err(SafeNativeError::InvariantViolation)?;
+    let mut out = String::new();
+    native_format_impl(context, gas_params.base, &ty, v, &mut out)?;
+    let move_str = Value::struct_(Struct::pack(vec![Value::vector_u8(out.into_bytes())]));
+    Ok(smallvec![move_str])
+}
+
+pub fn native_format_list(
+    gas_params: &GasParameters,
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut arguments: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    debug_assert!(ty_args.len() == 1);
+    let mut list_ty = &ty_args[0];
+
+    let arg_mismatch = 1;
+    let invalid_fmt = 2;
+
+    let val = safely_pop_arg!(arguments, Reference);
+    let mut val = val
+        .read_ref()
+        .map_err(SafeNativeError::InvariantViolation)?;
+
+    let fmt = safely_pop_arg!(arguments, Reference);
```

Comment:
You can actually coerce this by reference avoiding a copy, as [seen here](https://github.com/aptos-labs/aptos-core/blob/a1ae60836f987cacb6ed3f731e585df7c684c67e/third_party/move/move-stdlib/src/natives/string.rs#L127).

Reply:
Done